lfr/7pin_cardinal_demo/tests

```diff
@@ -0,0 +1,54 @@
+[Tests]
+  [generate_mesh]
+    type = RunApp
+    executable_pattern = 'cardinal*'
+    input = 'NTmesh.i  HCmesh.i'
+    cli_args = '--mesh-only'
+  []
+  [syntax_griffin]
+    type = RunApp
+    input = 'NT.i'
+    check_input = true
+    executable_pattern = 'cardinal*'
+    prereq = generate_mesh
+    max_time = 400
+  []
+  [syntax_conduction]
+    type = RunApp
+    input = 'HC.i'
+    check_input = true
+    executable_pattern = 'cardinal*'
+    prereq = generate_mesh
+  []
+  [syntax_nekrs]
+    type = RunApp
+    input = 'nek.i'
+    check_input = true
+    executable_pattern = 'cardinal*'
+  []
+
+  [run_multiphysics]
+    type = RunApp
```

Comment:
it would be better to have a CSVDiff, we are not far from being able to run this. 144 is not very far from what we have now on rod